hail/src/main/scala/is/hail/utils/ArrayOfByteArrayOutputStream.scala

```diff
@@ -0,0 +1,57 @@
+package is.hail.utils
+
+import java.io.{ByteArrayOutputStream, OutputStream}
+
+class ArrayOfByteArrayOutputStream(initialBufferCapacity: Int) extends OutputStream {
+
+  val MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+
+  /**
+    * The buffer where data is stored.
+    */
+  protected var buf = new BoxedArrayBuilder[ByteArrayOutputStream](1)
+  buf += new ByteArrayOutputStream(initialBufferCapacity)
+
+  protected var bytesInCurrentArray = 0
+  protected var currentArray = 0
```

Comment:
instead of having a currentArray index, can we just have a `currentBuilder: ByteArrayOutputStream`? I think having to do a lookup by index in the BoxedArrayBuilder in `buffer` might be a performance hit, and the code will probably be simpler regardless.